### PR TITLE
Bugfix: Add logging and testing to apply offsets in the Correction Loop

### PR DESCRIPTION
**Pull request by mmaclay**
The main objective of this PR is to fix an edge case bug where some parameters (e.g. hps.az_ang_nonlin, etc) were quietly not being updated correctly with applied offsets, including in the tests. 

The other focus is making the parameter application process more transparent and robust. This pull request introduces logging and testing for parameter offset application in the correction pipeline. The logging makes for easier debugging and verification, and the tests cover expected parameter types and edge cases.

**Logging and Transparency Enhancements:**

* Added detailed logging of generated parameter sets, including a summary table with parameter values and unit conversions for verification in `load_param_sets`.
* Improved logging in `apply_offset` to show before/after statistics (mean values and deltas) for modified fields, unit conversions performed, and warnings for missing fields or columns.
* Added logging of parameter details (type, field, value, units) during kernel creation in `_create_parameter_kernels` for better traceability.

**Testing:**

* Added a comprehensive test function `test_apply_offset_function` covering all parameter types (`OFFSET_KERNEL`, `OFFSET_TIME`, `CONSTANT_KERNEL`), unit conversions, negative offsets, missing fields, pass-through behavior, and ensuring data is not modified in place. Also verifies preservation of DataFrame columns.

These changes greatly improve the maintainability and reliability of the correction module by making parameter application steps clear and testable.

### CODE REVIEW

**Review by maxinelasp — CHANGES_REQUESTED** [review]

File: curryer/correction/correction.py [hunk 1]
```diff
@@ -1475,7 +1475,51 @@ def load_param_sets(config: CorrectionConfig) -> [ParameterConfig, typing.Any]:
             out_set.append((param, param_vals))
```
[the remark this answers is not in the document]
Comment (mmaclay) [reply curryer/correction/correction.py:1475]: added a test for this - using the passed value in seconds directly.

File: tests/test_correction/test_correction.py [hunk 2]
```diff
@@ -1827,13 +1827,15 @@ def test_apply_offset_function(self):
             ),
         )
 
-        # Apply time offset of 10 milliseconds
+        # Apply time offset - value must be in seconds
+        # This simulates the production flow where load_param_sets converts to seconds
         offset_ms = 10.0
+        offset_seconds = offset_ms / 1000.0  # Convert to internal units (seconds)
```
[the remark this answers is not in the document]
Comment (mmaclay) [reply tests/test_correction/test_correction.py:1833]: Added a test to show that load_param_sets passes seconds, and conversion is handled.

**Review by maxinelasp — APPROVED** [review]
